packages/embark/src/cmd/cmd_controller.js

```diff
@@ -504,58 +500,67 @@ class EmbarkController {
     const isSecondaryProcess = (engine) => { return engine.ipc.connected && engine.ipc.isClient(); };
 
     async.waterfall([
-      function initEngine(callback) {
+      callback => {
         engine.init({}, callback);
       },
-      function startServices(callback) {
+      callback => {
         let pluginList = engine.plugins.listPlugins();
         if (pluginList.length > 0) {
           engine.logger.info(__("loaded plugins") + ": " + pluginList.join(", "));
         }
 
-        engine.startService("web3");
-        engine.startService("deployment");
-        engine.startService("codeGenerator");
-        engine.startService("codeRunner");
-        engine.startService("console");
+        engine.registerModuleGroup("coreComponents");
+        engine.registerModuleGroup("stackComponents");
 
+        engine.registerModuleGroup("blockchain");
+        engine.registerModuleGroup("compiler");
+        engine.registerModuleGroup("contracts");
+        engine.registerModuleGroup("pipeline");
+        engine.registerModuleGroup("filewatcher");
+        engine.registerModuleGroup("storage");
+        engine.registerModuleGroup("communication");
+        engine.registerModulePackage('embark-deploy-tracker', {plugins: engine.plugins});
+        engine.registerModulePackage('embark-plugin-cmd', {embarkConfigFile: engine.embarkConfig, embarkConfig: engine.config.embarkConfig, packageFile: 'package.json'});
+        callback();
+      },
+      callback => {
         if (isSecondaryProcess(engine)) {
           return callback();
         }
-        engine.startService("embarkListener");
-        engine.startService("processManager");
-        engine.startService("coreProcess");
-        engine.startService("serviceMonitor");
-        engine.startService("libraryManager");
-        engine.startService("pipeline");
-        engine.startService("storage");
-        engine.startService("cockpit");
-        engine.startService("pluginCommand");
 
-        engine.events.request('blockchain:ready', callback);
-      },
-      function ipcConnect(callback) {
-        return callback();
-      },
-      function deploy(callback) {
-        // Skip if we are connected to a websocket, the server will do it
-        if(isSecondaryProcess(engine)) {
-          return callback();
-        }
-        engine.config.reloadConfig();
-        engine.events.request('deploy:contracts', function (err) {
-          callback(err);
+        let plugin = engine.plugins.createPlugin('cmdcontrollerplugin', {});
+
+        plugin.registerActionForEvent("embark:engine:started", async (_params, cb) => {
+          await engine.events.request2("blockchain:node:start", engine.config.blockchainConfig);
+          try {
+            await Promise.all([
+              engine.events.request2("storage:node:start", engine.config.storageConfig),
+              engine.events.request2("communication:node:start", engine.config.communicationConfig)
```

Comment:
Made concurrent as discussed. Notice that I have to spin up `blockchain:node:start` first though due to race conditions. `web3` otherwise won't be available in the other request routines.